unstract/core/src/unstract/core/pubsub_helper.py

```diff
@@ -96,9 +91,22 @@ def log_prompt(
     @classmethod
     def publish(cls, channel_id: str, payload: dict[str, Any]) -> bool:
         channel = f"logs:{channel_id}"
+        logs_expiry = int(os.environ.get("LOGS_EXPIRATION_TIME_IN_SECOND", 3600))
         try:
-            cls.r.publish(channel, json.dumps(payload))
+            log_data = json.dumps(payload)
+            cls.r.publish(channel, log_data)
+
+            # Check if the payload type is "LOG"
+            if payload["type"] == "LOG":
```

Comment:
It should n't be here. please move this in to consumer in backend.
Also, we have some changes in consumer too for design change of socket IO